zilliqa/src/db.rs

```diff
@@ -883,13 +883,13 @@ pub fn checkpoint_block_with_state<P: AsRef<Path> + Debug>(
             .at_root(bincode::deserialize::<Account>(&serialised_account)?.storage_root);
         let mut account_storage_buf = vec![];
         for (storage_key, storage_val) in account_storage.iter() {
-            account_storage_buf.extend_from_slice(&(storage_key.len() as u16).to_be_bytes());
+            account_storage_buf.extend_from_slice(&u16::try_from(storage_key.len())?.to_be_bytes());
```

Comment:
Aren't storage keys fixed length, just like account keys and thus the length here is unnecessary?

Reply:
Its storage keys which do have the variable length for Scilla, as you suspected the account keys might in the original bug ticket